src/reflect/scala/reflect/internal/AnnotationInfos.scala

```diff
@@ -58,7 +58,7 @@ trait AnnotationInfos extends api.Annotations { self: SymbolTable =>
     @tailrec private
     def dropOtherAnnotations(anns: List[AnnotationInfo], cls: Symbol): List[AnnotationInfo] = anns match {
```

Comment:
isnt this just a while loop, that can avoid the unapplySeq and pattern match?

Reply:
There is no `unapplySeq` call for case class constructor patterns, and tailrec emits a loop for this:

```
scala> class Tail { def count[A](as: List[A], z: Int = 0): Int = as match { case a :: bs => count(bs, z + 1); case _ => z }}
defined class Tail

scala> :javap -c Tail#count
  public <A> int count(scala.collection.immutable.List<A>, int);
    Code:
       0: aload_1
       1: astore        4
       3: aload         4
       5: instanceof    #19                 // class scala/collection/immutable/$colon$colon
       8: ifeq          38
      11: aload         4
      13: checkcast     #19                 // class scala/collection/immutable/$colon$colon
      16: astore        5
      18: aload         5
      20: invokevirtual #23                 // Method scala/collection/immutable/$colon$colon.tl$access$1:()Lscala/collection/immutable/List;
      23: astore        6
      25: aload_0
      26: aload         6
      28: iload_2
      29: iconst_1
      30: iadd
      31: invokevirtual #25                 // Method count:(Lscala/collection/immutable/List;I)I
      34: istore_3
      35: goto          46
      38: goto          41
      41: iload_2
      42: istore_3
      43: goto          46
      46: iload_3
      47: ireturn
```

But a while loop might be just as readable.